crates/cadenza-syntax/STATUS.md

```diff
@@ -0,0 +1,450 @@
+
+**Root Cause:** Parser marker propagation issue with low binding power operators inside delimiters.
+
+When parsing `{ x = 1 }`:
+1. Parser enters `parse_record` with `BraceMarker`
+2. Parses field with `CommaMarker(BraceMarker)`
+3. Sees `=` operator (binding power 5,4)
+4. Creates new `WhitespaceMarker` for RHS, **loses delimiter context**
+5. After parsing `1`, sees `}` on same line
+6. Whitespace marker says "continue" 
+7. Juxtaposition (BP 6) >= 4, tries to consume `}` as argument
+8. Error: closing brace consumed incorrectly
+
+**Solution:** Propagate parent delimiter markers through recursive parsing so child contexts know when to stop.
+
+**Workaround:** Shorthand expansion (`{ x }` → `{ x = x }`) deferred to macro expansion time.
+
+**References:** `PARSER_ISSUES.md` Issue 5
+
+---
+
+### 6. Record Field Access ✅ COMPLETE
+
+**Status:** Dot notation fully implemented.
+
+**Working:**
+- Simple access: `point.x`
+- Chained access: `obj.field.subfield`
+- After function calls: `get_point().x`
+- With indexing: `arr[0].name`
+- In expressions: `point.x + point.y`
+
+**Representation:** `Apply(., [record, field])`
+
+**Binding Power:** Field access (30, 31) - high precedence, left-associative
+
+**Test Files:** `op-field-*.cdz` in test-data/
+
+**References:** `PARSER_ISSUES.md` Issue 6
+
+---
+
+### 7. Tuples ❌ NOT IMPLEMENTED
+
+**Status:** Parentheses currently only used for grouping.
+
+**What's Needed:**
+1. Detect comma in parentheses to distinguish tuple from grouping
+2. Single element without comma: `(x)` is grouping
+3. Multiple elements or trailing comma: `(x, y)` or `(x,)` is tuple
+4. Represent as `Apply(__tuple__, [elements...])`
+5. Empty tuple: `()` → `Apply(__tuple__, [])`
+
+**Design Decision:** Should `()` be empty tuple or unit type?
+
+**References:** `PARSER_ISSUES.md` Issue 7
+
+---
+
+### 8. Match Expressions ❌ NOT IMPLEMENTED
+
+**Status:** No pattern matching support.
+
+**What's Needed:**
+1. Parse match arms with patterns and bodies
+2. Support `|` as pattern separator or indentation-based
+3. Represent as `Apply(match, [scrutinee, arm1, arm2, ...])`
+4. Each arm: `Apply(__arm__, [pattern, body])`
+5. Support pattern syntax: literals, identifiers, constructors
+6. Handle guard clauses: `pattern if condition -> body`
+
+**Syntax Examples:**
+```cadenza
+match x
+    0 -> "zero"
+    1 -> "one"
+    _ -> "other"
+```
+
+**References:** `PARSER_ISSUES.md` Issue 8
+
+---
+
+### 9. If/Else Expressions ❌ NOT IMPLEMENTED
+
+**Status:** No conditional expression support.
+
+**What's Needed:**
+1. Recognize `if condition then consequent else alternative` pattern
+2. Support indented bodies after `then` and `else`
+3. Represent as `Apply(if, [condition, consequent, alternative])`
+4. Support `elif` chains by nesting
+5. Handle if without else (if language design permits)
+
+**Syntax Examples:**
+```cadenza
+if x > 0 then "positive" else "negative"
+
+if condition then
+    do_something
+else
+    do_other
```

Comment:
Is there a syntax that we wouldn't have to specialize on for handling? Like this will force the parser to understand what `if` and `else` mean. The thing I really like about our current parser is there are only really only identifiers and punctuation; no keywords at all. The `match` syntax works quite well with this setup. We could also do a `cond` syntax:

```
cond
    expr1 -> 1
    expr2 -> 2
    true -> "base case"
```

But it would be nice to support `if` since that's what most people are familiar with. Any ideas about possible solutions?